### PR TITLE
[Flink-16083][chinese-translation]Translate "Dynamic Table" page of "…

### PR DESCRIPTION
**Pull request by CrazyTomatoOo**
…Streaming Concepts" into Chinese

## What is the purpose of the change

This pull request is to translate "Dynamic Table" page of "Streaming Concepts" into Chinese

## Verifying this change

This change is a trivial rework / code cleanup without any test coverage.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency):no
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: no
  - The serializers: no
  - The runtime per-record code paths (performance sensitive): no
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn/Mesos, ZooKeeper: no
  - The S3 file system connector: no
## Documentation

  - Does this pull request introduce a new feature? no
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)

### CODE REVIEW

File: docs/dev/table/streaming/dynamic_tables.zh.md [hunk 1]
```diff
@@ -159,31 +159,31 @@ FROM (
 );
 {% endhighlight %}
 
-The [Query Configuration](query_configuration.html) page discusses parameters to control the execution of continuous queries. Some parameters can be used to trade the size of maintained state for result accuracy.
+[查询配置](query_configuration.html)章节讨论了控制连续查询执行的参数。一些参数可以用来更改维持状态的大小以获得结果的准确性。
```
Comment (libenchao) [inline docs/dev/table/streaming/dynamic_tables.zh.md]: 一些参数可以用来更改维持状态的大小以获得结果的准确性 -> 一些参数可以用来在维持状态的大小和获得结果的准确性之间做取舍。

File: docs/dev/table/streaming/dynamic_tables.zh.md [hunk 2]
```diff
@@ -159,31 +159,31 @@ FROM (
...
 
-Table to Stream Conversion
+表到流的转换
 --------------------------
 
-A dynamic table can be continuously modified by `INSERT`, `UPDATE`, and `DELETE` changes just like a regular database table. It might be a table with a single row, which is constantly updated, an insert-only table without `UPDATE` and `DELETE` modifications, or anything in between.
+动态表可以像普通数据库表一样通过 `INSERT`、`UPDATE` 和 `DELETE` 来不断修改。它可能是一个只有一行、不断更新的表，也可能是一个 insert-only 的表，没有 `UPDATE` 和 `DELETE` 修改，或者介于两者之间的其他修改。
```
Comment (libenchao) [inline docs/dev/table/streaming/dynamic_tables.zh.md]: 或者介于两者之间的其他修改 -> 或者介于两者之间的其他表

File: docs/dev/table/streaming/dynamic_tables.zh.md [hunk 3]
```diff
@@ -159,31 +159,31 @@ FROM (
...
 
 <center>
 <img alt="Dynamic tables" src="{{ site.baseurl }}/fig/table-streaming/undo-redo-mode.png" width="85%">
 </center>
 <br><br>
 
-* **Upsert stream:** An upsert stream is a stream with two types of messages, *upsert messages* and *delete messages*. A dynamic table that is converted into an upsert stream requires a (possibly composite) unique key. A dynamic table with unique key is converted into a stream by encoding `INSERT` and `UPDATE` changes as upsert messages and `DELETE` changes as delete messages. The stream consuming operator needs to be aware of the unique key attribute in order to apply messages correctly. The main difference to a retract stream is that `UPDATE` changes are encoded with a single message and hence more efficient. The following figure visualizes the conversion of a dynamic table into an upsert stream.
+* **Upsert 流:** upsert 流包含两种类型的 message： *upsert messages* 和*delete messages*。转换为 upsert 流的动态表需要(可能是组合的)惟一键。通过将 `INSERT` 和 `UPDATE` 操作编码为 upsert message，将 `DELETE` 操作编码为 delete message ，将具有唯一键的动态表转换为流。消费流的算子需要知道唯一键的属性，以便正确地应用 message。与 retract 流的主要区别在于 `UPDATE` 操作是用单个 message 编码的，因此效率更高。下图显示了将动态表转换为 upsert 流的过程。
```
Comment (libenchao) [inline docs/dev/table/streaming/dynamic_tables.zh.md]: 惟一 -> 唯一

**Comment by CrazyTomatoOo** [comment]
Hi @libenchao , thanks for your comments. I have corrected them.

**Comment by libenchao** [comment]
LGTM. +1 for merging.

File: docs/dev/table/streaming/dynamic_tables.zh.md [hunk 4]
```diff
@@ -159,31 +159,31 @@ FROM (
 );
 {% endhighlight %}
 
-The [Query Configuration](query_configuration.html) page discusses parameters to control the execution of continuous queries. Some parameters can be used to trade the size of maintained state for result accuracy.
+[查询配置](query_configuration.html)章节讨论了控制连续查询执行的参数。一些参数可以用来在维持状态的大小和获得结果的准确性之间做取舍。
 
-Table to Stream Conversion
+表到流的转换
```
Comment (wuchong) [inline docs/dev/table/streaming/dynamic_tables.zh.md]: I only add a <a> html tag above this to make the anchor link work. 

```html
<a name="table-to-stream-conversion"></a>
```

This is an missing tip in the [Translation Specifications](https://cwiki.apache.org/confluence/display/FLINK/Flink+Translation+Specifications), I just updated the documentation.

File: docs/dev/table/streaming/dynamic_tables.zh.md [hunk 5]
```diff
@@ -22,135 +22,135 @@ specific language governing permissions and limitations
...
+虽然这两个示例查询看起来非常相似(都计算分组计数聚合)，但它们在一个重要方面不同:
+- 第一个查询更新先前输出的结果，即定义结果表的 changelog 流包含 `INSERT` 和 `UPDATE` 操作。
+- 第二个查询只附加到结果表，即结果表的 changelog 流只包含 `INSERT` 操作。
 
-Whether a query produces an append-only table or an updated table has some implications:
-- Queries that produce update changes usually have to maintain more state (see the following section).
-- The conversion of an append-only table into a stream is different from the conversion of an updated table (see the [Table to Stream Conversion](#table-to-stream-conversion) section).
+一个查询是产生一个只追加的表还是一个更新的表有一些含义:
+- 产生更新更改的查询通常必须维护更多的状态(请参阅以下部分)。
+- 将 append-only 的表转换为流与将已更新的表转换为流是不同的(参阅 [表到流的转换](#table-to-stream-conversion)章节)。
```
Comment (wuchong) [inline docs/dev/table/streaming/dynamic_tables.zh.md]: The anchor link is broken because we translate the title into Chinese.

**Pull request closed without merging**